File: software/examples/java/ExamplePrintVoltages.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickHAT;
import com.tinkerforge.BrickHAT.Voltages;

public class ExamplePrintVoltages {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XXYYZZ to the UID of your HAT Brick
	private static final String UID = "XXYYZZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickHAT hat = new BrickHAT(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current get voltages
		Voltages voltages = hat.getVoltages(); // Can throw com.tinkerforge.TimeoutException

		System.out.println("Voltage USB: " + voltages.voltageUSB/1000.0 + " V");
		System.out.println("Voltage DC: " + voltages.voltageDC/1000.0 + " V");

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/javascript/ExamplePrintVoltages.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XXYYZZ'; // Change XXYYZZ to the UID of your HAT Brick

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var hat = new Tinkerforge.BrickHAT(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current get voltages
        hat.getVoltages(
            function (voltageUSB, voltageDC) {
                console.log('Voltage USB: ' + voltageUSB/1000.0 + ' V');
                console.log('Voltage DC: ' + voltageDC/1000.0 + ' V');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
